lib/msal-common/src/authority/Authority.ts

```diff
@@ -653,23 +737,41 @@ export class Authority {
         let metadata = this.getCloudDiscoveryMetadataFromConfig();
         if (metadata) {
             this.logger.verbose(
-                "Found cloud discovery metadata in the config."
+                "Found cloud discovery metadata in authority configuration"
             );
             metadataEntity.updateCloudDiscoveryMetadata(metadata, false);
             return AuthorityMetadataSource.CONFIG;
         }
 
-        // If the cached metadata came from config but that config was not passed to this instance, we must go to the network
+        // If the cached metadata came from config but that config was not passed to this instance, we must go to hardcoded values
+        this.logger.verbose(
+            "Did not find cloud discovery metadata in the config... Attempting to get cloud discovery metadata from the hardcoded values."
+        );
+
+        const hardcodedMetadata =
+            this.getCloudDiscoveryMetadataFromHardcodedValues();
+        if (hardcodedMetadata && !this.options.skipAuthorityMetadataCache) {
+            this.logger.verbose(
+                "Found cloud discovery metadata from hardcoded values."
+            );
+            metadataEntity.updateCloudDiscoveryMetadata(
+                hardcodedMetadata,
+                false
+            );
+            return AuthorityMetadataSource.HARDCODED_VALUES;
+        }
+
         this.logger.verbose(
-            "Did not find cloud discovery metadata in the config... Attempting to get cloud discovery metadata from the cache."
+            "Did not find cloud discovery metadata in hardcoded values... Attempting to get cloud discovery metadata from the cache."
```

Comment:
Not entirely accurate, we may have hardcoded metadata, but the user has configured the option to skip it.  The logs won't represent that state as this is written.

Reply:
You're absolutely right, updated with an extra check and log message to be accurate. Thanks for catching this!